packages/react-storage/src/components/StorageBrowser/views/LocationActionView/UploadView/useUploadView.ts

```diff
@@ -3,30 +3,58 @@ import React from 'react';
 import { uploadHandler } from '../../../actions';
 
 import { useGetActionInput } from '../../../providers/configuration';
-import { useStore } from '../../../providers/store';
+import { FileItems, useStore } from '../../../providers/store';
 import { Task, useProcessTasks } from '../../../tasks';
 
-import { DEFAULT_ACTION_CONCURRENCY } from '../constants';
+import {
+  DEFAULT_ACTION_CONCURRENCY,
+  UPLOAD_FILE_SIZE_LIMIT,
+} from '../constants';
 import { UploadViewState, UseUploadViewOptions } from './types';
 import { DEFAULT_OVERWRITE_ENABLED } from './constants';
+import { isUndefined } from '@aws-amplify/ui';
 
 export const useUploadView = (
   options?: UseUploadViewOptions
 ): UploadViewState => {
+  const [invalidFiles, setInvalidFiles] = React.useState<
+    FileItems | undefined
+  >();
   const { onExit: _onExit } = options ?? {};
   const getInput = useGetActionInput();
   const [{ files, location }, dispatchStoreAction] = useStore();
   const { current, key } = location;
 
+  const validFiles = React.useMemo(
+    () =>
+      files?.filter((fileItem) => {
+        const { id, file } = fileItem;
+        if (file.size > UPLOAD_FILE_SIZE_LIMIT) {
+          setInvalidFiles((prev) =>
+            isUndefined(prev) ? [fileItem] : prev.concat(fileItem)
+          );
+          dispatchStoreAction({ type: 'REMOVE_FILE_ITEM', id });
```

Comment:
Nit: I'd avoid attach a side effect into the filter callback function.

Reply:
Taking a closer look at this one and I realize if we have multiple files larger than 160 GB, the side effects `setInvalidFiles` and `dispatchStoreAction` will not work as we wanted. So instead of supporting adding files to the file context and remove invalid ones in this hook, we should prevent adding these invalid files to the state in the 1st place.